Packages/com.unity.inputsystem/InputSystem/Actions/Composites/Vector2Composite.cs

```diff
@@ -67,17 +67,37 @@ public class Vector2Composite : InputBindingComposite<Vector2>
         /// vectors will have a magnitude > 1 (i.e. will be <c>new Vector2(1,1)</c>, for example,
         /// instead of <c>new Vector2(1,1).normalized</c>).
         /// </summary>
+        /// <value>Whether the normalize the resulting vector.</value>
         public bool normalize = true;
 
+        /// <summary>
+        /// If true (default), the up/down/left/right inputs will be treated as analog meaning that
+        /// their actual values will be used. If this is false, the values will be read as buttons
+        /// meaning that instead if an input value is below the button press threshold, the value
+        /// that is used is 0 and otherwise the value that is used is 1.
+        /// </summary>
+        /// <value>Whether to treat part bindings as analog controls.</value>
+        public bool analog = true;
+
```

Comment:
1. Are we ok with changing the default behavior here?
2. Does it make sense to have a normalize and an analog flag? If normalize is on, that kind of cancels out the effect of analog. Is it a desirable option to have analog and normalized enabled? Or should we have an enum with three modes (analog, digital, digital-normalized)?

Reply:
>Are we ok with changing the default behavior here?

Was deliberating with myself, too. Leaning towards yes. Reasoning:

1) The current default behavior has caught users by surprise.
2) The vast majority of existing bindings are things like WASD and they are unaffected.
3) Restoring the old behavior is easy.

>Does it make sense to have a normalize and an analog flag? If normalize is on, that kind of cancels out the effect of analog. Is it a desirable option to have analog and normalized enabled? Or should we have an enum with three modes (analog, digital, digital-normalized)?

Hmm, good question. The enum mode toggle sounds like a good idea. Agree that going with analog makes the value of the normalize option (especially as it's on by default) questionable.